### PR TITLE
Feature/show voice actor on actor detailspage

### PR DESCRIPTION
**Pull request by rikardomarenzzi**

### CODE REVIEW

File: backend/src/main/java/de/neuefische/rikardo/dub/service/DbService.java [hunk 1]
```diff
@@ -20,8 +23,29 @@ public DbService(VoiceActorMongoDb voiceActorMongoDb, ActorService actorService)
...
+
+        for (VoiceActor voiceActor : voiceActors) {
+            VoiceActorPreview voiceActorPreview = new VoiceActorPreview(
+                    voiceActor.getId(),
+                    voiceActor.getName(),
+                    voiceActor.getImage(),
+                    voiceActor.getType());
+            voiceActorPreviews.add(voiceActorPreview);
+        }
+        return voiceActorPreviews;
```
Comment (jamarob) [inline backend/src/main/java/de/neuefische/rikardo/dub/service/DbService.java:39]: Maybe use StreamingAPIs map here?

File: backend/src/test/java/de/neuefische/rikardo/dub/controller/TmdbControllerIntegrationTest.java [hunk 2]
```diff
@@ -35,24 +38,46 @@
...
+            new ActorPreview("13548","James Spader","https://image.tmdb.org/t/p/w154/image.jpg","actor")
+    ));
+    List<VoiceActor> voiceActors = new ArrayList<>(List.of(
+            new VoiceActor("1","Dietmar Wunder","/dietmar_wunder.jpeg","1965-12-05",actorPreviewsWunder,"voiceactor"),
+            new VoiceActor("2","Benjamin Völz","/benjamin_voelz.jpeg","1960-05-13",actorPreviewsVoelz,"voiceactor")
+    ));
+
+    VoiceActor voiceActor = new VoiceActor("1","Dietmar Wunder","/dietmar_wunder.jpeg","1965-12-05",actorPreviews,"voiceactor");
+
+    Actor actor = new Actor("6384","Keanu Reeves","https://image.tmdb.org/t/p/w154/image.jpg","Neo","biography","1964-09-02","Beirut, Lebanon","actor",moviePreviews,voiceActorPreviews);
```
Comment (jamarob) [inline backend/src/test/java/de/neuefische/rikardo/dub/controller/TmdbControllerIntegrationTest.java:79]: Maybe remove the unused test data?

File: backend/src/test/java/de/neuefische/rikardo/dub/service/ActorServiceTest.java [hunk 3]
```diff
@@ -30,17 +34,47 @@
...
+    List<VoiceActorPreview> voiceActorPreviewVoelz = new ArrayList<>(List.of(
+            new VoiceActorPreview("2","Benjamin Völz","/benjamin_voelz.jpeg","voiceactor")
+    ));
+
+    List<VoiceActorPreview> voiceActorPreviewWunder  = new ArrayList<>(List.of(
+            new VoiceActorPreview("1","Dietmar Wunder","/dietmar_wunder.jpeg","voiceactor")
+    ));
+
+    Actor actor = new Actor("6384","Keanu Reeves","https://image.tmdb.org/t/p/w154/image.jpg","Neo","biography","1964-09-02","Beirut, Lebanon","actor",moviePreviews,voiceActorPreviewVoelz);
+
```
Comment (jamarob) [inline backend/src/test/java/de/neuefische/rikardo/dub/service/ActorServiceTest.java:70]: Maybe you can remove some duplication by extracting your test data into its own util class.

File: backend/src/test/java/de/neuefische/rikardo/dub/service/ActorServiceTest.java [hunk 4]
```diff
@@ -30,17 +34,47 @@
...
 
 
     @Test
-    @DisplayName("The method should return a valid URL to search for a actor by name")
+    @DisplayName("The method should return an actorpreview by name")
```
Comment (jamarob) [inline backend/src/test/java/de/neuefische/rikardo/dub/service/ActorServiceTest.java:74]: This is nitpicky, but the 'This method' is just noise that can be deleted to get short and sweet test descriptions.

File: frontend/src/detailsPage/ActorPage.js [hunk 5]
```diff
@@ -7,10 +7,19 @@ export default function ActorPage() {
   const history = useHistory();
   const [actor] = useActor();
 
+
   return (
     <>
       <div>{actor?.name}</div>
```
Comment (jamarob) [inline frontend/src/detailsPage/ActorPage.js:13]: Heading? if so use an heading tag ;)

File: frontend/src/detailsPage/ActorPage.js [hunk 6]
```diff
@@ -7,10 +7,19 @@ export default function ActorPage() {
...
       <img alt="ActorImage" src={actor?.image}/>
```
Comment (jamarob) [inline frontend/src/detailsPage/ActorPage.js:14]: alt strings are there to transport information, when the image can not be loaded, the name of the actor would be better.

File: frontend/src/detailsPage/ActorPage.js [hunk 7]
```diff
@@ -7,10 +7,19 @@ export default function ActorPage() {
...
+
+      {actor?.voiceActors.length > 0 &&
+        <>
+      <div>{actor?.voiceActors[0].name}</div>
+      <img alt="VoiceActorImage" src={actor?.voiceActors[0].image} onClick={onImageClick} height="231px" width="154px"/>
+      </>
```
Comment (jamarob) [inline frontend/src/detailsPage/ActorPage.js:20]: You already checked that `actor` is not nullish in line 16, no need to repeat it in this block.

Again, the alt string of the image should tell what is visible on the image if that information is not available elsewhere on the page, can probably be set to an empty string here as you display the actors name.

The image sizes seem oddly specific?

File: frontend/src/detailsPage/ActorPage.js [hunk 8]
```diff
@@ -7,10 +7,19 @@ export default function ActorPage() {
...
+      }
+
       <button onClick={onCancel}>Cancel</button>
       <label>Birthday<p>{actor?.birthday}</p></label>
```
Comment (jamarob) [inline frontend/src/detailsPage/ActorPage.js:24]: labels are mostly used for inputs, there is a <time> tag, that might be suitable for the birthday.

File: frontend/src/detailsPage/ActorPage.js [hunk 9]
```diff
@@ -7,10 +7,19 @@ export default function ActorPage() {
...
+
   return (
     <>
       <div>{actor?.name}</div>
       <img alt="ActorImage" src={actor?.image}/>
+
+      {actor?.voiceActors.length > 0 &&
+        <>
+      <div>{actor?.voiceActors[0].name}</div>
+      <img alt="VoiceActorImage" src={actor?.voiceActors[0].image} onClick={onImageClick} height="231px" width="154px"/>
```
Comment (jamarob) [inline frontend/src/detailsPage/ActorPage.js:19]: Why are you only interested in the first VoiceActor?

**Review by jamarob — COMMENTED** [review]
Seems like it is really coming together! 🎉 

Try to take some time to clean up unused code, also beware of variables that where named after their type, but the type has changed, this might lead to confusion.

Try to use semantic tags in the frontend whereever see https://allthetags.com/ for reference.

**Comment by rikardomarenzzi** [comment]
Thank you very much for your detailed review. In the following PR i'm going to take the comments into account.

**Pull request merged**